### PR TITLE
712 - `{shinytest2}` for `tm_g_distribution`

### PR DESCRIPTION
**Pull request by vedhav**
Part of #712

### CODE REVIEW

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  app$set_module_input("strata_i-dataset_ADSL_singleextract-filter1-vals", c("USA", "CAN", "BRA"))
+  app$set_module_input("bins", 50)
+  app$set_module_input("add_dens", FALSE)
+
+  # Encodings in the QQplot tab
+  app$set_module_input("tabs", "QQplot")
+  app$set_module_input("t_dist", "normal")
+  app$click(NS(app$active_module_ns(), "params_reset"))
+
+  app$expect_no_shiny_error()
```
Comment (averissimo) [inline tests/testthat/test-shinytest2-tm_g_distribution.R:98]: Do you think it's equivalent to do this just once?

**Review by averissimo — COMMENTED** [review]
Are you testing the `data_extract_vars` here?

File: tests/testthat/test-shinytest2-tm_g_distribution.R (hunk 1 above)
Comment (vedhav) [reply tests/testthat/test-shinytest2-tm_g_distribution.R:98]: It should have been a `app$expect_no_validation_error()` as the test message indicates. I changed it.
I'm just testing if the encoding inputs create output without a validation error. Irrespective of what type of inputs they are. Not setting all the inputs, setting just the main ones.

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  testthat::expect_null(app$get_active_module_input("t_dist"))
+  testthat::expect_identical(app$get_active_module_input("dist_param1"), NA)
+  testthat::expect_identical(app$get_active_module_input("dist_param2"), NA)
+  testthat::expect_null(app$get_active_module_input("dist_tests"))
+  testthat::expect_equal(app$get_active_module_input("roundn"), 2)
+  testthat::expect_equal(app$get_active_module_input("ggtheme"), "gray")
+  testthat::expect_equal(app$get_active_module_input("tabs"), "Histogram")
+
+  # Encodings in the QQplot tab
+  app$set_module_input("tabs", "QQplot")
```
[the remark this answers is not in the document]
Comment (m7pr) [reply tests/testthat/test-shinytest2-tm_g_distribution.R:73]: wow

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_distribution()
+
+  # Encodings in the Histogram tab
+  app$set_module_input("group_i-dataset_ADSL_singleextract-filter1-col", "ARM")
+  app$set_module_input("group_i-dataset_ADSL_singleextract-filter1-vals", c("A: Drug X", "C: Combination"))
+  app$set_module_input("scales_type", "Free")
+  app$set_module_input("strata_i-dataset_ADSL_singleextract-filter1-col", "COUNTRY")
+  app$set_module_input("strata_i-dataset_ADSL_singleextract-filter1-vals", c("USA", "CAN", "BRA"))
```
Comment (m7pr) [inline tests/testthat/test-shinytest2-tm_g_distribution.R:89]:
```suggestion
  app$set_active_module_input("strata_i-dataset_ADSL_singleextract-filter1-vals", c("USA", "CAN", "BRA"))
```

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  app$set_module_input("bins", 50)
```
Comment (m7pr) [inline tests/testthat/test-shinytest2-tm_g_distribution.R:90]:
```suggestion
  app$set_active_module_input("bins", 50)
```

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  app$set_module_input("add_dens", FALSE)
```
Comment (m7pr) [inline tests/testthat/test-shinytest2-tm_g_distribution.R:91]:
```suggestion
  app$set_active_module_input("add_dens", FALSE)
```

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+  # Encodings in the QQplot tab
+  app$set_module_input("tabs", "QQplot")
```
Comment (m7pr) [inline tests/testthat/test-shinytest2-tm_g_distribution.R:94]:
```suggestion
  app$set_active_module_input("tabs", "QQplot")
```

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  app$set_module_input("t_dist", "normal")
```
Comment (m7pr) [inline tests/testthat/test-shinytest2-tm_g_distribution.R:95]:
```suggestion
  app$set_active_module_input("t_dist", "normal")
```

**Review by m7pr — CHANGES_REQUESTED** [review]
rename `set_module_input` to `set_active_module_input`

File: tests/testthat/test-shinytest2-tm_g_distribution.R [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  app_driver$set_active_module_input("strata_i-dataset_ADSL_singleextract-filter1-vals", c("USA", "CAN", "BRA"))
+  app_driver$set_active_module_input("bins", 50)
+  app_driver$set_active_module_input("add_dens", FALSE)
+
+  # Encodings in the QQplot tab
+  app_driver$set_active_module_input("tabs", "QQplot")
+  app_driver$set_active_module_input("t_dist", "normal")
+  app_driver$click(NS(app_driver$active_module_ns(), "params_reset"))
+
+  app_driver$expect_no_validation_error()
```
[the remark this answers is not in the document]
Comment (averissimo) [reply tests/testthat/test-shinytest2-tm_g_distribution.R:118]: Solved in 87e7a534

**Review by averissimo — APPROVED** [review]
LGTM!

**Pull request merged**